crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -1148,5 +1148,185 @@ static_assert(not is_subtype_of(TypeOf[A.g], Callable[[], int]))
 static_assert(is_subtype_of(TypeOf[A.f], Callable[[A, int], int]))
 ```
 
+### Overloads
+
+#### Subtype overloaded
+
+For `B <: A`, if a callable `B` is overloaded with two or more signatures, it is a subtype of
+callable `A` if _at least one_ of the overloaded signatures in `B` is a subtype of `A`.
+
+`overloaded.pyi`:
+
+```pyi
+from typing import overload
+
+class A: ...
+class B: ...
+class C: ...
+
+@overload
+def overloaded(x: A) -> None: ...
+@overload
+def overloaded(x: B) -> None: ...
+```
+
+```py
+from knot_extensions import CallableTypeOf, is_subtype_of, static_assert
+from overloaded import A, B, C, overloaded
```

Comment:
I'm curious why we are using multiple models and an import here? The thing we are testing doesn't seem related to imports; this could all be in one file. (Same question for similar cases below.)

Reply:
Do you mean why I've defined it in a stub file? It's mainly so that I don't need to add the implementation function and making sure that it's consistent with the overload signatures. Otherwise I'd need to make sure that they are consistent or I could use protocols which is yet to be implemented or abstract methods which would make the test setup a bit verbose (add `@abstractmethod` to all methods).

Happy to answer if the query is related to something else.

Reply:
No makes sense! Failed to process that it was a stub file